packages/core/src/agent/Agent.ts

```diff
@@ -141,6 +141,10 @@ export class Agent<AgentModules extends AgentModulesInput = any> extends BaseAge
     this.messageSender.registerOutboundTransport(outboundTransport)
   }
 
+  public async resetOutboundTransport() {
+    await this.messageSender.resetOutboundTransport()
+  }
```

Comment:
hmm good point. Maybe that method could then also remove all as a special case

Reply:
I'd keep it simple. The need for unregistering it special and needing to remove them one by one is fine in this case (otherwise the naming of the method will be harder again)

Reply:
fair enough